### DISCUSSION TITLE
这种点X和整个页面右边都没变化的页面怎么定位呀？

### DISCUSSION THREAD

**Discussion by dearmb** [gkd-kit/gkd#768] [General]
https://i.gkd.li/snapshot/1731169552091
这种点X和整个页面右边都没变化的页面怎么定位呀？
青椒

**Comment by lisonge** [comment]
你没有生成快照分享链接，其它设备打开链接只会得到快照缺失的提示

![image](https://github.com/user-attachments/assets/51bce0b9-1081-4f65-a9b9-f8faee79f16a)

**Comment by dearmb** [comment]
[snapshot-1731169552091.zip](https://github.com/user-attachments/files/17691520/snapshot-1731169552091.zip)
附件可以吗？分享要好多条件，好麻烦呀

**Comment by lisonge** [comment]
内部是一个整体的网页，没有提供什么无障碍信息，只能坐标定位，而且可能容易误触

<https://i.gkd.li/i/17692372?gkd=V2ViVmlld1t0ZXh0PSLmlK_ku5jkvJjmg6DljaEiXSA8IFdlYlZpZXcgPCBGcmFtZUxheW91dCA8IFt2aWQ9InJlZnJlc2hMYXlvdXQiXQ>

然后点击固定坐标 `909, 743` 或者使用控件宽度自己计算

**Reply by dearmb** [reply]
gkd怎么坐标定位呢？你说他那个整体网页，X为啥也能正常点

**Reply by lisonge** [reply]
https://gkd.li/api/interfaces/RawRuleProps#position

**Reply by dearmb** [reply]
点击鼠标能在审查工具里测试吗？

**Reply by dearmb** [reply]
找了个例子，但是没看明白怎么应用在我这个需求上
 {
                        "key": 2,
                        "fastQuery": true,
                        "position": {
                            "left": "width * 0.9",
                            "top": "width * 0.1676"
                        },
                        "matches": "[id=\"com.UCMobile.adn_jingdong_sdk:id/animation_view\"]",
                        "exampleUrls": "https://e.gkd.li/92da64ad-5b08-4343-bec2-bb6a7e0bb92d",
                        "snapshotUrls": "https://i.gkd.li/i/16494884"
                    }

**Reply by dearmb** [reply]
"matches": "[id="com.UCMobile.adn_jingdong_sdk:id/animation_view"]",
"exampleUrls": "https://e.gkd.li/92da64ad-5b08-4343-bec2-bb6a7e0bb92d",
"snapshotUrls": "https://i.gkd.li/i/16494884"

这三行不需要吗？

**Reply by lisonge** [reply]
> 点击鼠标能在审查工具里测试吗？

这个无法测试，因为不知道怎么测试

**Reply by lisonge** [reply]
```json5
{
  id: 'com.xunlei.downloadprovider',
  name: '迅雷',
  groups: [
    {
      key: 1,
      name: '[ChangeMe]规则名称-2024-11-11 22:12:46',
      desc: '[ChangeMe]本规则由GKD网页端审查工具生成',
      rules: [
        {
          fastQuery: true,
          activityIds: 'com.xunlei.web.XLWebViewActivity$XLWebViewActivityImpl2',
          matches: 'WebView[text="支付优惠卡"] < WebView < FrameLayout < [vid="refreshLayout"]',
"position": {
"left": "909",
"top": "743"
},
          exampleUrls: 'https://e.gkd.li/5101a0d6-b940-466d-906d-904bba9aa50b',
          snapshotUrls: 'https://i.gkd.li/i/17692372',
        },
      ],
    },
  ],
}
```

**Reply by dearmb** [reply]
谢谢，可以导入，这个广告出现不多，下次看看成功不成功吧
还有两个问题请教一下
1，我记得李跳跳好像是无感点击，就是好像点击的很快，没看到广告就关掉了，GKD好像还要看一下广告闪一下再关闭，能优化吗？也或许是我记错了，李跳跳也是闪一下
2，审查工具咋生成的这个规则呀，我没找到按钮

**Reply by lisonge** [reply]
> 审查工具咋生成的这个规则呀，我没找到按钮

手写的

---
> GKD好像还要看一下广告闪一下再关闭

在规则过多或没有开启快速查询的情况下会出现这个情况，你可以尝试使用

https://gkd.li/api/interfaces/RawCommonProps#prioritytime

**Reply by dearmb** [reply]
我还以为写的审查工具生成，是审查工具有生成规则的按钮

prioritytime看了一下想想算了，感觉对于我来说学习成本有点大，搞搞基本的就得了
